client/app/lib/hooks/useAddToRecentsAtLocalStorage.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from "react";
+
+export default function useAddToRecentsAtLocalStorage(id, localStorageProperty) {
+  function updateList (list, id) {
+    const itemArrayIndex = list.indexOf(id);
+    const itemIsAlreadyOnTheList = itemArrayIndex !== -1;
+    if(itemIsAlreadyOnTheList)
+      list.splice(itemArrayIndex, 1);
+    list.unshift(id);
+    return list;
+  }
+
+  return useCallback(() => {
```

Comment:
I guess that is important for this feature to also cover the "execute" event, so maybe the useEffect ins't the best choice. But i agree that this can be a regular JS function, i will apply this change, but i'm kinda confused about the project folder structure, where can i put this function file?